### PR TITLE
[SPARK-49211][SQL][3.5] V2 Catalog can also support built-in data sources

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

V2 Catalog can also support built-in data sources.

### Why are the changes needed?

V2 catalog could still support spark built-in data sources if the V2 catalog returns v1 table and do not track partitions in catalog. This is because we do not need to require V2 catalog to implement every thing to support built-in data sources (as that is a big chunk of work).

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

UT
### Was this patch authored or co-authored using generative AI tooling?

NO

### CODE REVIEW

**Comment by amaliujia** [comment]
@cloud-fan

**Comment by HyukjinKwon** [comment]
Is this a backport? or 3.5 specific issue?

**Comment by amaliujia** [comment]
https://github.com/apache/spark/pull/47723 

@HyukjinKwon we merged this to master but need a different PR for 3.5 due to merge conflict.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
```
DataSourceV2SQLSuiteV1Filter.SPARK-49211: V2 Catalog support CTAS
java.lang.NullPointerException: null
```

@amaliujia can you look into the test failures?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/identifiers.scala [hunk 1]
```diff
@@ -111,6 +111,10 @@ case class QualifiedTableName(database: String, name: String) {
   override def toString: String = s"$database.$name"
 }
 
+case class FullQualifiedTableName(catalog: String, database: String, name: String) {
```
Comment (pan3793) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/identifiers.scala:114]: should comments for `QualifiedTableName` be updated?

**Comment by cloud-fan** [comment]
thanks, merging to 3.5!

**Pull request closed without merging**